core/cloudflow-flink/src/main/scala/cloudflow/flink/FlinkStreamlet.scala

```diff
@@ -80,24 +80,64 @@ abstract class FlinkStreamlet extends Streamlet[FlinkStreamletContext] with Seri
-   * by [[createExecutionEnvironment]].
-   * @see [[createExecutionEnvironment]] which creates the StreamExecutionEnvironment used in this FlinkStreamlet.
+   * Populate Flink configuration from Streamlet Flink configuration
    */
-  protected def setupExecutionEnvironment(env: StreamExecutionEnvironment): StreamExecutionEnvironment = env
+  private def populateFlinkConfiguration(configuration: Configuration, config: Config, path: String): Boolean = {
+    var enableWeb = false
+    config
+      .atPath(path)
+      .entrySet()
+      .forEach {
+        // According to https://ci.apache.org/projects/flink/flink-docs-stable/ops/config.html
+        // Values can be Int, bool, duratio and String
+        // We are not supporting Duration here
+        entry =>
+          // Remove the path from the key
+          val key = entry.getKey.replace(path, "")
+          entry.getValue.valueType() match {
+            case ConfigValueType.BOOLEAN => configuration.setBoolean(key, entry.getValue.asInstanceOf[Boolean])
+            case ConfigValueType.NUMBER  => configuration.setInteger(key, entry.getValue.asInstanceOf[Int])
+            case _                       => configuration.setString(key, entry.getValue.toString)
+          }
+          if (key == ClusterFlinkJobExecutor.enableLocalWeb) enableWeb = true;
+      }
+    enableWeb
+  }
 
   /**
    * Creates the Flink StreamExecutionEnvironment and by default sets up exactly-once checkpointing.
    * @see [[setupExecutionEnvironment]] to modify the StreamExecutionEnvironment that is created by this method.
    */
-  protected def createExecutionEnvironment: StreamExecutionEnvironment = {
-    val env = StreamExecutionEnvironment.getExecutionEnvironment
+  protected def createExecutionEnvironment(config: Config, streamlet: String): StreamExecutionEnvironment = {
+
+    val localMode = config.as[Option[Boolean]]("cloudflow.local").getOrElse(false)
+    val env = localMode match {
+      case true =>
+        val configuration = new Configuration()
+        // Copy all of the Flink parameters to configuration
+        val enableWeb = populateFlinkConfiguration(configuration, config, ClusterFlinkJobExecutor.flinkRuntime) ||
+            populateFlinkConfiguration(configuration, config, ClusterFlinkJobExecutor.streamletRuntimeConfigPath(streamlet))
+        // Ensures that filesystem is initialized with right configuration
+        FileSystem.initialize(configuration, null)
+        // Create local Flink environment
+        // Note here that a local web support is set through configuration by setting
+        // "local.web" either in the streamlet or Flink runtime context. We are not checking the value, but rather
+        // the fact that variable is defines.
+        enableWeb match {
+          case true => StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(configuration)
+          case _    => StreamExecutionEnvironment.createLocalEnvironment(1, configuration)
+        }
```

Comment:
nitpick: I would do an if-then-else here. Pattern matching is probably more expensive

Reply:
Looks more "scalish" to me, but I might be wrong

Reply:
I think it's only really scala-ish if it's some kind of union type, with a simple boolean I think if is more common

Reply:
You guys are so hard to please, done